src/Ouroboros/Network/Protocol/UntypedStream/Server.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+module Ouroboros.Network.Protocol.UntypedStream.Server where
+
+import Protocol.Untyped
+import Pipes (Producer)
+import qualified Pipes
+
+import Ouroboros.Network.Protocol.UntypedStream.Type
+
+newtype UntypedStreamServer m rng a t = UntypedStreamServer {
```

Comment:
What is the `rng`?

Reply:
`rng` stands for `range`, as in ask for a `range` of blocks / block bodies.